packages/common/src/types/ccdAmount.ts

```diff
@@ -6,7 +10,29 @@
 export class CcdAmount {
     microCcdAmount: bigint;
 
-    constructor(microCcdAmount: bigint) {
+    /**
+     * Constructs a CcdAmount and checks that it is valid. It accepts a number, string, big, or bigint as parameter.
+     * It can accept a string as parameter with either a comma or a dot as the decimal separator.
+     *
+     * @param microCcdAmount The amount of micro CCD as a number, string, big, or bigint.
+     * @throws If a number is passed with several decimal seperators
+     * @throws If a negative amount of micro CCD is passed
+     * @throws If the micro CCD passed is greater than what can be contained in a 64-bit integer
```

Comment:
We could add here that it also throws if passed a non-whole number. I think we should consider consolidating all these @throws declarations into a single one such as:

```suggestion
     * @throws If an invalid micro CCD amount is passed, i.e. any value which is not an unsigned 64-bit integer
```